packages/@icon-magic/generate/src/generate-worker.ts

```diff
@@ -100,12 +112,47 @@ async function generateSingleIcon(iconConfig: IconConfig): Promise<void> {
  */
 async function applyGeneratePluginsOnFlavors(
   icon: Icon,
-  plugins: GeneratePlugin[]
+  plugins: GeneratePlugin[],
+  type: RegExp,
+  hashing?: boolean
 ): Promise<Flavor[]> {
   let promises: Flavor[] = [];
   if (icon.flavors) {
     for (const iconFlavor of icon.flavors.values()) {
       LOGGER.debug(`Applying plugins on ${icon.iconName}'s ${iconFlavor.name}`);
+      // Check if generate has been run on this flavor already, if it has, it will be saved
+      // in the iconrc in the output path
+      if (hashing) {
+        // Check if generate has been run on this flavor already
+        const flavorName: string = path.basename(iconFlavor.name);
+        // Create the output directory
+        const outputPath = icon.getIconOutputPath();
+        // Find the flavors in the config from the initial run that match the flavorName
+        const savedFlavorConfigs = await hasAssetBeenProcessed(
+          outputPath,
+          flavorName,
+          iconFlavor,
+          type
+        );
+        if (savedFlavorConfigs && savedFlavorConfigs.length) {
+          // Make flavors from the already written config
+          savedFlavorConfigs.forEach(
+            async (savedFlavorConfig: FlavorConfig) => {
+              // Create new Flavor from the config we retrieved, so it's copied over
```

Comment:
Can we create a new icon itself from the config that we've read instead of having to navigate through the JSON and then creating flavors manually